src/components/Common/SearchByMultipleFields.tsx

```diff
@@ -177,6 +177,18 @@ const SearchByMultipleFields: React.FC<SearchByMultipleFieldsProps> = ({
     return () => document.removeEventListener("keydown", handleKeyDown);
   }, [focusedIndex, open, handleOptionChange, options]);
 
+  useEffect(() => {
+    if (autoFocus) {
+      inputRef.current?.focus();
+    }
+  }, [autoFocus, open, selectedOptionIndex]);
```

Comment:
why was this moved to the top? let's keep the diff clean. avoid unnecessary changes. do a self-review